### PR TITLE
Added an example of parametrized query

### PR DESCRIPTION
**Pull request by meiriko**
This addition adds the person type to show how to do a parametrized query with a computed field and infinite drill down (by using the resolver).

### CODE REVIEW

**Comment by DxCx** [comment]
Hi @meiriko Thanks!
i'll take a deeper look later,
jut wanted to let you know the tests are failing because of a rouge space :\
```diff
-   "testString": "it Works!",
+   "testString": "it Works! ",
```

File: src/schema/modules/query.ts [hunk 1]
```diff
@@ -6,13 +8,17 @@ type Query {
...
 `;
 
 export const resolver = {
     Query: {
+        getPerson(root, args, ctx) {
+            return findPerson(args.id);
+        },
         testString() {
-            return "it Works!";
+            return "it Works! ";
```
Comment (DxCx) [inline src/schema/modules/query.ts]: there. :)

File: src/schema/modules/person-type.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        id: 3,
+        sex: 'male',
+        name: 'joe'
+    }
+];
+
+export const resolver = {
+    PersonType: {
+        matches(root, args, context) {
+            return persons.filter(person => person.sex !== root.sex);
```
Comment (DxCx) [inline src/schema/modules/person-type.ts]: Please add persons from ctx objects to match real world case :)
also i would move it into a new file of person-database.ts or something like that.

File: src/schema/modules/person-type.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        }
+    },
+};
+
+export const findPerson = (id: number) => {
```
Comment (DxCx) [inline src/schema/modules/person-type.ts]: i would move this function to the same external file..

**Comment by DxCx** [comment]
Looks great generally! thanks!
i have some small comments but i really like it :)

**Comment by meiriko** [comment]
Excellent points. Ignore this branch, I'll find time to fix the tests, write test to my additions and create a new pull requests. You see, contributing to the community is a way to benefit, now I have to conform to writing tests.
Can you please direct me at the documentation regarding the adding from ctx object ?

**Comment by DxCx** [comment]
When i create the server on main, i pass the context object. Which then provided to the resolvers.

If not clear leave it as is and then ill fix it later and point you towards it.

If you dont have time for tests thats fine (i'll add later), i meant that you added a space which broke an existing test..

**Comment by meiriko** [comment]
I'll do it, no touch :-) That's a learning opportunity

**Comment by meiriko** [comment]
Okay. I re factored the files and added tests. Thank you for a great and helpful feedback. Please scrutinize the new pull request, it is an opportunity for me to improve.

**Comment by meiriko** [comment]
Also I noticed that there is a conflict in main.ts though I only added stuff (added the context). I think if you merge (accept the changes) things should work fine.

File: src/schema/modules/query.ts [hunk 4]
```diff
@@ -15,10 +15,10 @@ type Query {
 export const resolver = {
     Query: {
         getPerson(root, args, ctx) {
-            return findPerson(args.id);
+            return findPerson(ctx.persons, args.id);
```
Comment (DxCx) [inline src/schema/modules/query.ts]: What you really want to do is:
```typescript
return ctx.findPerson(args.id);
```

File: src/schema.spec.ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import { Schema } from "./schema";
 import { graphql } from "graphql";
 import 'jest';
+import { persons } from "./schema/data-base/person-database";
```
Comment (DxCx) [inline src/schema.spec.ts]: please take data-base folder 1 directory up.
should be import from "./database/person-database";

File: src/main.ts [hunk 6]
```diff
@@ -35,7 +36,9 @@ export function main(options: IMainOptions) {
     }
 
     app.use(GRAPHQL_ROUTE, bodyParser.json(), graphqlExpress({
-        context: {},
+        context: {
+            persons
```
Comment (DxCx) [inline src/main.ts]: you can import findPerson in main and pass it here in context along with persons

File: src/schema/modules/person-type.ts [hunk 7]
```diff
@@ -7,32 +7,10 @@ type PersonType {
...
-    {
-        id: 3,
-        sex: 'male',
-        name: 'joe'
-    }
-];
-
 export const resolver = {
     PersonType: {
         matches(root, args, context) {
```
Comment (DxCx) [inline src/schema/modules/person-type.ts]: can you please rename "context" to "ctx" to match the rest of the files?

Comment (DxCx) [inline src/schema/modules/person-type.ts]: wow nice!
ok great :)

**Comment by meiriko** [comment]
Ahhh! Here's the rub (Hamlet). Not that proficient with rebasing. Do I do it on top of my master? What is the command?

**Comment by DxCx** [comment]
hmm.
you will first need to add my github as a remote (`git remote add github git@github.com:DxCx/webpack-apollo-server.git`)
then pull latest master to your master (`git checkout master && git remote update github && git merge -no-ff github/master`)
*you will probably encounter conflicts, you should run `git mergetool` to resolve*
finally, push it. `git push origin master`

**Comment by meiriko** [comment]
Done. Note that jest-cli was missing, added it to devDependencies

**Comment by DxCx** [comment]
Thanks alot!

**Comment by meiriko** [comment]
Thank you. I Learned a lot. More to come...

**Pull request merged**